### PR TITLE
Implement Input::release_all function

### LINKED ISSUE #7297: Input actions don't update pressed states after OS window adjustment
Operating system: Windows 8.1(64)
Godot version: Godot_v2.1.1-stable_win64
Issue description: 
Input actions don't update pressed states after OS window adjustment. This may create an input saying its pressed even though you released it outside of the game.

Steps to reproduce:
1.) press and hold an action
2.) adjust OS window and release action 

Video of me doing it:
[https://youtu.be/3bZuJVrmGJw](https://youtu.be/3bZuJVrmGJw)

### LINKED ISSUE #19137: Persistent Input key pressed Bug/Glitch
**Godot version:**
3.0.2

**OS/device including version:**
Windows 7 x64 RX 470

**Issue description:**
Input is Stuck

**Steps to reproduce:**
Greetings everyone, I would just like to share that I've notice some bug and there are 3 ways to reproduce this issue. For example if you are player and is moving using move_and_slide or position movement.

If Godot or your computer freezes or hangs for a bit, the input is stuck and is being read as key pressed. Another way to reproduce this is when you drag around the window of running godot game while moving(which makes it stutter), the Input also ends up stuck. Lastly, the easiest way to reproduce this issue is when you rapidly press shift+key (which enables sticky key for windows), The input key pressed ends up stuck.

Also, just wanted to follow up, there are cases that when if godot stutters/hangs, it seems to skip some frames/cycles of the code and causes inaccurate calculation which I'm not sure if its normal.

### LINKED ISSUE #22265: Need to press F5/F6 twice on 3.x in release_debug
I see this #19576 closed, i open new one because bug not fixed.

**Godot version:**
All 3.x,  master 0de6514

**OS/device including version:**
Xubuntu 16.04.5 LTS, Nvidia GTX 750 Ti
Tested 3.x on Windows 7 on the same machine and it works.
I think is only affected x11.

**Issue description:**
Sometimes need to press F5 or F6 twice to run the project or scene with Godot 3.x.

**Steps to reproduce:**
1) Open scene.
2) Press F6 button and hold it and release after scene is run. 
3) Close windows and press F6.
4) Scene is not play, need press F6 twice.
5) Try several times it's difficult to reproduce.
 
On my machine this happens randomly not necessary to hold F6 button.

**Minimal reproduction project:**
Simple project with one 2DNode for version 3.1.
[f6.zip](https://github.com/godotengine/godot/files/2398814/f6.zip)

After compare 2.1 version with 3.x, i found what cause this behaviour on my machine.
https://github.com/godotengine/godot/blob/e20a28a7db090dbf39bc8c90487c3d7beb84cbb8/main/input_default.cpp#L266-L272

Sometimes keys_pressed.erase not reached, then **last_is_pressed == true**, and **k->set_echo(true)**
Because **BaseButton::_unhandled_input** verify if not **is_echo()**, scene not run, while key not pressed twice.. 
**New code from 3.x:**
https://github.com/godotengine/godot/blob/e20a28a7db090dbf39bc8c90487c3d7beb84cbb8/platform/x11/os_x11.cpp#L1684-L1694

### PR DESCRIPTION
**Pull request by marcelofg55**
Fixes #7297
Fixes #19137
Fixes  #22265

### CODE REVIEW

File: main/input_default.cpp [hunk 1]
```diff
@@ -580,6 +580,15 @@ void InputDefault::action_release(const StringName &p_action) {
...
 }
 
+void InputDefault::action_release_all() {
+
+	for (Map<StringName, Action>::Element *E = action_state.front(); E; E = E->next()) {
+		if (E->get().pressed) {
+			action_release(E->key());
+		}
+	}
+}
```
Comment (allkhor) [inline main/input_default.cpp]:
```suggestion
	keys_pressed.clear();
}
```
If we also clear keys_pressed, PR can fix #22265, WDYT?

Comment (marcelofg55) [reply main/input_default.cpp]: Sounds good, just added it, thanks!

Comment (bojidar-bg) [reply main/input_default.cpp]: Shouldn't it clear `joy_buttons_pressed` and `_joy_axis` in this case as well?

**Comment by reduz** [comment]
I don't think this is a good idea, specially if we are going to add multiple window support soon (which includes popups and other stuff that have complex multi window behaviors).
I think these issues are better handled on a case by case basis, you already get a WM focus out function to deal with issues.

**Comment by marcelofg55** [comment]
> I don't think this is a good idea, specially if we are going to add multiple window support soon (which includes popups and other stuff that have complex multi window behaviors).
> I think these issues are better handled on a case by case basis, you already get a WM focus out function to deal with issues.

How do we deal with #7297 then?

File: main/input_default.cpp (hunk 1 above)
Comment (marcelofg55) [reply main/input_default.cpp]: Just changed it to `release_all` and added the clears for `joy_buttons_pressed` and `_joy_axis`

**Comment by akien-mga** [comment]
I suggest to reevaluate this for 3.2 as both issues it addresses are not critical.

**Comment by akien-mga** [comment]
Superseded by #28061.

**Pull request closed without merging**